source/slang/slang-ir-glsl-legalize.cpp

```diff
@@ -3220,7 +3254,9 @@ void tryReplaceUsesOfStageInput(
                     }
                     else
                     {
-                        use->set(materializedInner);
+                        auto tmpVar = builder.emitVar(materializedInner->getDataType());
+                        builder.emitStore(tmpVar, materializedInner);
+                        use->set(tmpVar);
```

Comment:
It's a little terrifying seeing cases where the number of levels of indirection seems to have been changed (the use was previously getting replaced with a *value* even though the value being used is seemingly a *pointer*).

These are exactly the kinds of type mismatches that would be great to have an IR validation pass catch.